src/frontend/src/scheduler/distributed/query.rs

```diff
@@ -229,6 +237,12 @@ impl QueryExecution {
     }
 }
 
+impl Drop for QueryExecution {
+    fn drop(&mut self) {
+        self.query_metrics.distributed_running_query.dec();
+    }
+}
```

Comment:
Currently the `QueryExecution` will not be removed immediately, it's better to put the decrement logic in `QueryRunner` when it finished.